api/app.go

```diff
@@ -419,6 +424,9 @@ func updateApp(w http.ResponseWriter, r *http.Request, t auth.Token) (err error)
 	if updateData.Router != "" {
 		wantedPerms = append(wantedPerms, permission.PermAppUpdateRouter)
 	}
+	if updateData.UpdatePlatform {
+		wantedPerms = append(wantedPerms, permission.PermPlatformUpdate)
```

Comment:
`PermPlatformUpdate` is actually used by admins to update platforms installed in tsuru (we should not use it here). We can add a new permission, e.g 'PermAppImageReset', to check here.